py/selenium/webdriver/support/expected_conditions.py

```diff
@@ -95,6 +95,8 @@ def __call__(self, ignored):
 
 
 def _element_if_visible(element, visibility=True):
+    if isinstance(element, str) or isinstance(element, dict):
+           raise StaleElementReferenceException("Invalid locator")
```

Comment:
I've tested the code you wrote on a simple web page. What you wrote seems like it should work, except in the case of waiting for an element to go 'invisible' you should not care about the result (The result will be True for invisibility). Maybe you should go to the #selenium chat room on freenode IRC and ask for help there.